src/Temporalio/Workflow/WorkflowAttribute.cs

```diff
@@ -0,0 +1,246 @@
+                        {
+                            errs.Add($"WorkflowRun on {method} and {runMethod}");
+                        }
+                        else if (!method.IsPublic)
+                        {
+                            errs.Add($"WorkflowRun on {method} is not public");
+                        }
+                        else if (!typeof(Task).IsAssignableFrom(method.ReturnType))
+                        {
+                            errs.Add($"WorkflowRun method {method} must return an instance of Task");
+                        }
+                        else if (initConstructor != null &&
+                            !method.GetParameters().SequenceEqual(initConstructor.GetParameters()))
+                        {
+                            errs.Add($"WorkflowRun on {method} must match parameters of WorkflowInit on {initConstructor}");
+                        }
+                        else
+                        {
+                            runMethod = method;
+                        }
+                    }
+                    if (method.IsDefined(typeof(WorkflowSignalAttribute), false))
+                    {
+                        try
+                        {
+                            var defn = WorkflowSignalAttribute.Definition.FromMethod(method);
+                            if (signals.ContainsKey(defn.Name))
+                            {
+                                errs.Add($"{type} has more than one signal named {defn.Name}");
+                            }
+                            signals[defn.Name] = defn;
+                        }
+                        catch (ArgumentException e)
+                        {
+                            errs.Add(e.Message);
+                        }
+                    }
+                    if (method.IsDefined(typeof(WorkflowQueryAttribute), false))
+                    {
+                        try
+                        {
+                            var defn = WorkflowQueryAttribute.Definition.FromMethod(method);
+                            if (queries.ContainsKey(defn.Name))
+                            {
+                                errs.Add($"{type} has more than one query named {defn.Name}");
+                            }
+                            queries[defn.Name] = defn;
+                        }
+                        catch (ArgumentException e)
+                        {
+                            errs.Add(e.Message);
+                        }
+                    }
+                }
+                if (runMethod == null)
+                {
+                    errs.Add($"{type} does not have a valid WorkflowRun method");
+                }
+
+                // If there are any errors, throw
+                if (errs.Count > 0)
+                {
+                    // TODO(cretz): Ok to use aggregate exception here or should I just
+                    // comma-delimit into a single message or something?
```

Comment:
Seems like that's what it's there for

Reply:
Kinda, but it's not "AggregateMessageException", it has separate exceptions carrying their own stack traces, causes, etc. But I agree, probably good enough, and bonus is the outputters know how to format this one well.